### PR TITLE
[FSSDK-9972]  fix dependabot moderate alerts

### PR DESCRIPTION
**Pull request by pulak-opti**
## Summary
- updated dependencies to fix [dependabot moderate alerts](https://github.com/optimizely/agent/security/dependabot)
- github.com/lestrrat-go/jwx library introduced breaking changes to fix the vulnerability, so updated accordingly

## Ticket
- [FSSDK-9972](https://jira.sso.episerver.net/browse/FSSDK-9972)

### CODE REVIEW

File: pkg/middleware/auth.go [hunk 1]
```diff
@@ -130,15 +131,15 @@ func (c *JWTVerifierURL) updateKeySet() error {
 	c.jwksLock.Lock()
 	defer c.jwksLock.Unlock()
 
-	set, err := jwk.Fetch(c.jwksURL)
+	set, err := jwk.Fetch(context.TODO(), c.jwksURL)
```
Comment (mikechu-optimizely) [inline pkg/middleware/auth.go:134]: I'm totally not knowledgeable, but `.TODO` caught my eye. This is correct right?

**Review by mikechu-optimizely — APPROVED** [review]
Small thing that caught my eye 👇 but LGTM

**Review by raju-opti — APPROVED** [review]
LGTM

File: pkg/middleware/auth.go (hunk 1 above)
Comment (pulak-opti) [reply pkg/middleware/auth.go:134]: Yes, for now, we have to pass a context there. TODO() is for in future we should pass the caller context here if possible.

**Pull request merged**